src/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,141 @@
+import classnames from 'classnames';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const METHODS_TO_BIND = [
+  'getIdealPosition',
+  'handleMouseEnter',
+  'handleMouseLeave'
+];
+
+class Tooltip extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {isOpen: false};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getIdealPosition() {
+    // Get the anchor and position from state if possible. If not, get it from
+    // the props.
+    let anchor = this.state.anchor || this.props.anchor;
+    let position = this.state.position || this.props.position;
+
+    if (this.refs.tooltipContent) {
+      let viewportHeight = global.window.innerHeight;
```

Comment:
There's a `DOMUtil#getViewportHeight` that you can use. You can also create a `#getViewportWidth` too